node/node.go

```diff
@@ -107,6 +108,12 @@ func NewNode(config cfg.Config, privValidator *types.PrivValidator, clientCreato
 	}
 	consensusReactor := consensus.NewConsensusReactor(consensusState, fastSync)
 
+	// Transaction indexing
+	store := dbm.NewDB("tx_indexer", config.GetString("db_backend"), config.GetString("db_dir"))
+	indexer := txindexer.NewKV(store)
+	bcReactor.SetTxIndexer(indexer)
```

Comment:
Then we don't need to pass it to ApplyBlock since the state will already have it